### DISCUSSION TITLE
Access Token and Button Links

### DISCUSSION THREAD

**Discussion by kjellhaaland** [directus/directus#7281] [Archive]
I want to create a button that opens a file in the browser when browsing the details of an item.
The desired output link is: http://localhost:8055/assets/FILE_UUID_HERE?access_token=ACCESS_TOKEN_HERE

I have tried two approaches without success:

**1 - By using _Button Links_.**
When using _Button Links_, there is no way of including the access token in the link.
Would it be challenging to expose the access token to the _Button Links_ component? 
Ex: https://example.com?access_token={{access_token}}

**2 - By creating a custom component button.**
When creating custom components, i cant figure out how to get the access_token to interact with Directus APIS.
Is there any documentation on how to achieve this?

**Comment by rijkvanzanten** [comment]
> Would it be challenging to expose the access token to the Button Links component?

Never thought of that before, but it makes total sense! We already have a 'wrap with current user access token' utility function that'll add the access token to a given link, so it should be fairly straightforward to add an option to the button links interface for `[ ] Add current user's token to url`. The only thing we'll have to keep in mind is that those tokens expire, and are connected to the current user 🤔
